src/main/java/org/jenkinsci/plugins/artifactdeployer/service/PermissionsUtils.java

```diff
@@ -0,0 +1,89 @@
+package org.jenkinsci.plugins.artifactdeployer.service;
+
+import java.io.File;
+import java.io.InputStream;
+import java.io.PrintStream;
+
+import org.jruby.ext.posix.POSIX;
+import org.jruby.ext.posix.POSIX.ERRORS;
+import org.jruby.ext.posix.POSIXFactory;
+import org.jruby.ext.posix.POSIXHandler;
```

Comment:
replace package with
jnr.posix
Currently you removed the dep to the atlassian ant task, but you still require the org.jruby.ext.posix stuff which is where the problem lies as this requires the outdated jna.
There is a replacement for this the jnr-posix which is also included by jenkins in the core.

Reply:
Thank you. I've made the change.